crypto/rand/drbg_lib.c

```diff
@@ -382,12 +382,20 @@ int rand_drbg_restart(RAND_DRBG *drbg,
     }
 
     /* repair error state */
-    if (drbg->state == DRBG_ERROR)
+    if (drbg->state == DRBG_ERROR) {
         RAND_DRBG_uninstantiate(drbg);
+        /* The drbg->ctr member needs to be reinitialized before reinstantiation */
+        RAND_DRBG_set(drbg, drbg->nid, drbg->flags);
+    }
 
     /* repair uninitialized state */
     if (drbg->state == DRBG_UNINITIALISED) {
-        drbg_setup(drbg, NULL, NULL);
```

Comment:
This was a hack, see commit message for https://github.com/openssl/openssl/pull/4402/commits/ff9e120e0d38d9e10ee498ad2d01bafca38bd55f - rand_drbg_restart: reinstantiate drbg without using drbg_setup() [fixup]